File: bls_jztc/demo/api/content/v1/category.go
package v1

import (
	"github.com/gogf/gf/v2/frame/g"
)

// HomeCategoryListReq 首页分类列表请求
type HomeCategoryListReq struct {
	g.Meta `path:"/home-category/list" method:"get" tags:"内容管理" summary:"获取首页分类列表" security:"Bearer" description:"获取首页分类列表，需要管理员权限"`
}

// HomeCategoryListRes 首页分类列表响应
type HomeCategoryListRes struct {
	g.Meta `mime:"application/json" example:"json"`
	List   []CategoryItem `json:"list" dc:"分类列表"`
}

// IdleCategoryListReq 闲置分类列表请求
type IdleCategoryListReq struct {
	g.Meta `path:"/idle-category/list" method:"get" tags:"内容管理" summary:"获取闲置分类列表" security:"Bearer" description:"获取闲置分类列表，需要管理员权限"`
}

// IdleCategoryListRes 闲置分类列表响应
type IdleCategoryListRes struct {
	g.Meta `mime:"application/json" example:"json"`
	List   []CategoryItem `json:"list" dc:"分类列表"`
}

// CategoryItem 分类项
type CategoryItem struct {
	Id        int    `json:"id" dc:"分类ID"`
	Name      string `json:"name" dc:"分类名称"`
	SortOrder int    `json:"sortOrder" dc:"排序值"`
	IsActive  bool   `json:"isActive" dc:"是否启用"`
	Icon      string `json:"icon" dc:"分类图标URL"`
	CreatedAt string `json:"createdAt" dc:"创建时间"`
	UpdatedAt string `json:"updatedAt" dc:"更新时间"`
}

// HomeCategoryCreateReq 创建首页分类请求
type HomeCategoryCreateReq struct {
	g.Meta    `path:"/home-category/create" method:"post" tags:"内容管理" summary:"创建首页分类" security:"Bearer" description:"创建首页分类，需要管理员权限"`
	Name      string `v:"required#分类名称不能为空" json:"name" dc:"分类名称"`
	SortOrder int    `v:"min:0#排序值不能小于0" json:"sortOrder" dc:"排序值"`
	IsActive  bool   `json:"isActive" dc:"是否启用"`
	Icon      string `json:"icon" dc:"分类图标URL"`
}

// HomeCategoryCreateRes 创建首页分类响应
type HomeCategoryCreateRes struct {
	g.Meta `mime:"application/json" example:"json"`
	Id     int `json:"id" dc:"分类ID"`
}

// IdleCategoryCreateReq 创建闲置分类请求
type IdleCategoryCreateReq struct {
	g.Meta    `path:"/idle-category/create" method:"post" tags:"内容管理" summary:"创建闲置分类" security:"Bearer" description:"创建闲置分类，需要管理员权限"`
	Name      string `v:"required#分类名称不能为空" json:"name" dc:"分类名称"`
	SortOrder int    `v:"min:0#排序值不能小于0" json:"sortOrder" dc:"排序值"`
	IsActive  bool   `json:"isActive" dc:"是否启用"`
	Icon      string `json:"icon" dc:"分类图标URL"`
}

// IdleCategoryCreateRes 创建闲置分类响应
type IdleCategoryCreateRes struct {
	g.Meta `mime:"application/json" example:"json"`
	Id     int `json:"id" dc:"分类ID"`
}

// HomeCategoryUpdateReq 更新首页分类请求
type HomeCategoryUpdateReq struct {
	g.Meta    `path:"/home-category/update" method:"put" tags:"内容管理" summary:"更新首页分类" security:"Bearer" description:"更新首页分类，需要管理员权限"`
	Id        int    `v:"required#分类ID不能为空" json:"id" dc:"分类ID"`
	Name      string `v:"required#分类名称不能为空" json:"name" dc:"分类名称"`
	SortOrder int    `v:"min:0#排序值不能小于0" json:"sortOrder" dc:"排序值"`
	IsActive  bool   `json:"isActive" dc:"是否启用"`
	Icon      string `json:"icon" dc:"分类图标URL"`
}

// HomeCategoryUpdateRes 更新首页分类响应
type HomeCategoryUpdateRes struct {
	g.Meta `mime:"application/json" example:"json"`
}

// IdleCategoryUpdateReq 更新闲置分类请求
type IdleCategoryUpdateReq struct {
	g.Meta    `path:"/idle-category/update" method:"put" tags:"内容管理" summary:"更新闲置分类" security:"Bearer" description:"更新闲置分类，需要管理员权限"`
	Id        int    `v:"required#分类ID不能为空" json:"id" dc:"分类ID"`
	Name      string `v:"required#分类名称不能为空" json:"name" dc:"分类名称"`
	SortOrder int    `v:"min:0#排序值不能小于0" json:"sortOrder" dc:"排序值"`
	IsActive  bool   `json:"isActive" dc:"是否启用"`
	Icon      string `json:"icon" dc:"分类图标URL"`
}

// IdleCategoryUpdateRes 更新闲置分类响应
type IdleCategoryUpdateRes struct {
	g.Meta `mime:"application/json" example:"json"`
}

// HomeCategoryDeleteReq 删除首页分类请求
type HomeCategoryDeleteReq struct {
	g.Meta `path:"/home-category/delete" method:"delete" tags:"内容管理" summary:"删除首页分类" security:"Bearer" description:"删除首页分类，需要管理员权限"`
	Id     int `v:"required#分类ID不能为空" json:"id" dc:"分类ID"`
}

// HomeCategoryDeleteRes 删除首页分类响应
type HomeCategoryDeleteRes struct {
	g.Meta `mime:"application/json" example:"json"`
}

// IdleCategoryDeleteReq 删除闲置分类请求
type IdleCategoryDeleteReq struct {
	g.Meta `path:"/idle-category/delete" method:"delete" tags:"内容管理" summary:"删除闲置分类" security:"Bearer" description:"删除闲置分类，需要管理员权限"`
	Id     int `v:"required#分类ID不能为空" json:"id" dc:"分类ID"`
}

// IdleCategoryDeleteRes 删除闲置分类响应
type IdleCategoryDeleteRes struct {
	g.Meta `mime:"application/json" example:"json"`
}

File: bls_jztc/jztc/src/api/category.ts
import request from '@/utils/request';

// 通用响应结构
export interface ApiResponse<T> {
  code: number;
  message: string;
  data: T;
}

// 分类项接口
export interface CategoryItem {
  id: number;
  name: string;
  isActive: boolean;
  sortOrder: number;
  icon?: string;
  createdAt: string;
  updatedAt: string;
}

// 首页分类列表响应
export interface HomeCategoryListRes {
  code: number;
  message: string;
  data: {
    list: CategoryItem[];
  };
}

// 创建首页分类请求参数
export interface HomeCategoryCreateReq {
  name: string;
  isActive?: boolean;
  sortOrder?: number;
  icon?: string;
}

// 创建首页分类响应
export interface HomeCategoryCreateRes {
  code: number;
  message: string;
  data: {
    id: number;
  };
}

// 更新首页分类请求参数
export interface HomeCategoryUpdateReq {
  id: number;
  name: string;
  isActive?: boolean;
  sortOrder?: number;
  icon?: string;
}

// 更新首页分类响应
export interface HomeCategoryUpdateRes {
  code: number;
  message: string;
  data: null;
}

// 删除首页分类响应
export interface HomeCategoryDeleteRes {
  code: number;
  message: string;
  data: null;
}

/**
 * 获取首页分类列表
 * @returns 首页分类列表
 */
export const getHomeCategoryList = (): Promise<HomeCategoryListRes> => {
  return request.get('/home-category/list');
};

/**
 * 创建首页分类
 * @param data 分类数据
 * @returns 创建结果
 */
export const createHomeCategory = (data: HomeCategoryCreateReq): Promise<HomeCategoryCreateRes> => {
  return request({
    url: '/home-category/create',
    method: 'post',
    data,
  });
};

/**
 * 更新首页分类
 * @param data 分类数据
 * @returns 更新结果
 */
export const updateHomeCategory = (data: HomeCategoryUpdateReq): Promise<HomeCategoryUpdateRes> => {
  return request({
    url: '/home-category/update',
    method: 'put',
    data,
  });
};

/**
 * 删除首页分类
 * @param id 分类ID
 * @returns 删除结果
 */
export const deleteHomeCategory = (id: number): Promise<HomeCategoryDeleteRes> => {
  return request.delete('/home-category/delete', { params: { id } });
};

// ===== 闲置分类相关接口 =====

// 闲置分类列表响应
export interface IdleCategoryListRes {
  code: number;
  message: string;
  data: {
    list: CategoryItem[];
  };
}

// 创建闲置分类请求参数
export interface IdleCategoryCreateReq {
  name: string;
  isActive?: boolean;
  sortOrder?: number;
  icon?: string;
}

// 创建闲置分类响应
export interface IdleCategoryCreateRes {
  code: number;
  message: string;
  data: {
    id: number;
  };
}

// 更新闲置分类请求参数
export interface IdleCategoryUpdateReq {
  id: number;
  name: string;
  isActive?: boolean;
  sortOrder?: number;
  icon?: string;
}

// 更新闲置分类响应
export interface IdleCategoryUpdateRes {
  code: number;
  message: string;
  data: null;
}

// 删除闲置分类响应
export interface IdleCategoryDeleteRes {
  code: number;
  message: string;
  data: null;
}

/**
 * 获取闲置分类列表
 * @returns 闲置分类列表
 */
export const getIdleCategoryList = (): Promise<IdleCategoryListRes> => {
  return request.get('/idle-category/list');
};

/**
 * 创建闲置分类
 * @param data 分类数据
 * @returns 创建结果
 */
export const createIdleCategory = (data: IdleCategoryCreateReq): Promise<IdleCategoryCreateRes> => {
  return request({
    url: '/idle-category/create',
    method: 'post',
    data,
  });
};

/**
 * 更新闲置分类
 * @param data 分类数据
 * @returns 更新结果
 */
export const updateIdleCategory = (data: IdleCategoryUpdateReq): Promise<IdleCategoryUpdateRes> => {
  return request({
    url: '/idle-category/update',
    method: 'put',
    data,
  });
};

/**
 * 删除闲置分类
 * @param id 分类ID
 * @returns 删除结果
 */
export const deleteIdleCategory = (id: number): Promise<IdleCategoryDeleteRes> => {
  return request.delete('/idle-category/delete', { params: { id } });
};

// ===== 所有分类相关接口 =====

/**
 * 所有分类项（包含分类类型）
 */
export interface CategoryWithType extends CategoryItem {
  type: 'home' | 'idle'; // home: 首页分类, idle: 闲置分类
}

/**
 * 获取所有分类响应
 */
export interface GetCategoriesRes {
  code: number;
  message: string;
  data: {
    homeCategories: CategoryItem[]; // 首页分类
    idleCategories: CategoryItem[]; // 闲置分类
  };
}

/**
 * 获取所有分类列表（包括首页和闲置）
 * @returns 所有分类列表
 */
export const getAllCategories = (): Promise<GetCategoriesRes> => {
  return request.get('/categories');
};

/**
 * 获取所有分类并转换为Select组件可用的格式
 * @returns 转换后的分类列表，用于Select组件
 */
export const getCategoriesForSelect = async (): Promise<{label: string, value: string, type: string}[]> => {
  const res = await getAllCategories();
  
  console.log('API返回的分类数据:', res);
  
  if (res.code === 0) {
    const { homeCategories, idleCategories } = res.data;
    
    // 合并并转换为Select组件可用的格式
    const options: {label: string, value: string, type: string}[] = [];
    
    // 添加首页分类
    if (homeCategories && homeCategories.length > 0) {
      options.push({ label: '首页分类', value: 'home-group', type: 'group' });
      homeCategories.forEach(item => {
        if (item.isActive) { // 只添加启用的分类
          options.push({
            label: item.name,
            value: `home:${item.name}`,
            type: 'home'
          });
        }
      });
    }
    
    // 添加闲置分类
    if (idleCategories && idleCategories.length > 0) {
      options.push({ label: '闲置分类', value: 'idle-group', type: 'group' });
      idleCategories.forEach(item => {
        if (item.isActive) { // 只添加启用的分类
          options.push({
            label: item.name,
            value: `idle:${item.name}`,
            type: 'idle'
          });
        }
      });
    }
    
    console.log('转换后的分类选项:', options);
    return options;
  }
  
  return [];
}; 